packages/x-charts/src/internals/material/index.tsx

```diff
@@ -0,0 +1,27 @@
+import { BaseButton } from './components/BaseButton';
+import { BaseIconButton } from './components/BaseIconButton';
+import type { ChartsBaseSlotProps, ChartsBaseSlots } from '../../models/slots/chartsBaseSlots';
+import type { ChartsIconSlotProps, ChartsIconSlots } from '../../models/slots/chartsIconSlots';
+import { NullComponent } from './components/NullComponent';
+
+import './augmentation';
+
+const baseSlots: ChartsBaseSlots = {
+  baseButton: BaseButton,
+  baseIconButton: BaseIconButton,
+
+  // Null components should be overridden by higher plan packages
+  baseDivider: NullComponent,
+  basePopper: NullComponent,
+  baseTooltip: NullComponent,
+  baseMenuItem: NullComponent,
+  baseMenuList: NullComponent,
```

Comment:
I'm a bit unsure about this. 

Before, pro-only components were defined in the pro package. However, since we use module augmentation for extending the props of these components, it would become messy to know if you should augment `@mui/x-charts` or `@mui/x-charts-pro` for every component's props. With this approach, users only need to augment `@mui/x-charts`. 

We could also define pro components' actual implementation here, but we would be increasing the bundle size of `@mui/x-charts` with components we know won't be used, so I decided for this `NullComponent` to reduce the impact on bundle size.

Let me know if you have any better suggestions.

Reply:
Shouldn't users be able to augment only the version they are using? So if the user is using pro, they should always use `declare module '@mui/x-charts-pro' { }` regardless of where the origin?